cohttp/src/transfer_io.ml

```diff
@@ -38,10 +38,11 @@ module Make (IO : S.IO) = struct
            > The chunked encoding allows each chunk to include zero or
            > more chunk extensions, immediately following the chunk-size
         *)
-        try String.sub chunk_size_hex 0 (String.index chunk_size_hex ';')
-        with _ -> chunk_size_hex
+        match String.index chunk_size_hex ';' with
+        | exception Not_found -> chunk_size_hex
+        | s -> String.sub chunk_size_hex 0 s
       in
-      try Some (Int64.of_string ("0x" ^ hex)) with _ -> None
+      Int64.of_string_opt ("0x" ^ hex)
```

Comment:
We can think about using the new parser within the http library as well:

```ocaml
match Http.Private.parse_chunk_length chunk_size_hex with
| Ok (chunk_length, consumed) -> assert (consumed = String.length chunk_size_hex); chunk_length
| Error Partial -> assert false (* this branch will never be reached here since we feed the full line *)
| Error (Msg msg) -> failwith msg
```

Reply:
I gave it a try. It's a big ugly because we need to add the newline at the end of the input.

Reply:
Hmm true. Forgot about the new line.

Reply:
It's still better this way I think. We can always tweak `parse_chunk_length` to not eat the newline